_overviews/core/collections-migration-213.md

```diff
@@ -0,0 +1,330 @@
+
+~~~ scala
+def mapInPlace(f: A => A): this.type
+def flatMapInPlace(f: A => IterableOnce[A]): this.type
+def filterInPlace(p: A => Boolean): this.type
+def patchInPlace(from: Int, patch: scala.collection.Seq[A], replaced: Int): this.type
+~~~
+
+Another new operation is `distinctBy`:
+
+~~~ scala
+def distinctBy[B](f: A => B): C // (Where `C` can be `List[Int]`, for instance)
+~~~
+
+Last, additional operations are provided by the `scala-collection-contrib` module. You can
+think of this artifact as an incubator: if we get evidence that these operations should be part of the core,
+we might eventually move them.
+
+The new operations are provided via an implicit enrichment. You need to add the following import to make them
+available:
+
+~~~ scala
+import strawman.collection.decorators._
+~~~
+
+The following operations are provided:
+
+- `Seq`
+    - `intersperse`
+- `Map`
+    - `zipByKey` / `join` / `zipByKeyWith`
+    - `mergeByKey` / `fullOuterJoin` / `mergeByKeyWith` / `leftOuterJoin` / `rightOuterJoin`
+
+## Are there new implementations of existing collection types (changes in performance characteristics)?
+
+The default `Set` and `Map` are backed by a `ChampHashSet` and a `ChampHashMap`, respectively. The performance characteristics are the same but the
+operation implementations are faster. These data structures also have a lower memory footprint.
+
+`mutable.Queue` and `mutable.Stack` now use `mutable.ArrayDeque`. This data structure supports constant time index access, and amortized constant time
+insert and remove operations.
+
+## How do I cross-build my project against Scala 2.12 and Scala 2.13?
+
+Most usages of collections are compatible and can cross-compile 2.12 and 2.13 (at the cost of some warnings, sometimes).
+
+If you cannot get your code to cross-compile, there are various solutions:
+  - You can maintain a separate branch with the changes for 2.13 and publish releases for 2.13 from this branch.
+  - You can put source files that don't cross-compile in separate directories and configure sbt to assemble the sources according to the Scala version (see also the examples below):
+
+        // Adds a `src/main/scala-2.13+` source directory for Scala 2.13 and newer
+        // and a `src/main/scala-2.13-` source directory for Scala version older than 2.13
+        unmanagedSourceDirectories in Compile += {
+          val sourceDir = (sourceDirectory in Compile).value
+          CrossVersion.partialVersion(scalaVersion.value) match {
+            case Some((2, n)) if n >= 13 => sourceDir / "scala-2.13+"
+            case _                       => sourceDir / "scala-2.13-"
+          }
+        }
+
+  - You can use the [`scala-collection-compat`](https://github.com/scala/scala-collection-compat) library, which makes some of 2.13's APIs available to 2.11 and 2.12. This solution does not always work, for example if your library implements custom collection types.
```

Comment:
I would order the solutions from the simplest to set up to the more complicated. Using `scala-collection-compat` should be the first one.

Reply:
👍